File: app/src/main/java/com/steevsapps/idledaddy/handlers/PurchaseResponse.java
package com.steevsapps.idledaddy.handlers;

import androidx.annotation.NonNull;

import com.steevsapps.idledaddy.handlers.callbacks.PurchaseResponseCallback;

import in.dragonbra.javasteam.base.ClientMsgProtobuf;
import in.dragonbra.javasteam.base.IPacketMsg;
import in.dragonbra.javasteam.enums.EMsg;
import in.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserver2;
import in.dragonbra.javasteam.steam.handlers.ClientMsgHandler;

public class PurchaseResponse extends ClientMsgHandler {
    @Override
    public void handleMsg(@NonNull IPacketMsg packetMsg) {
        if (packetMsg.getMsgType() == EMsg.ClientPurchaseResponse) {
            handlePurchaseResponse(packetMsg);
        }
    }

    private void handlePurchaseResponse(IPacketMsg packetMsg) {
        final ClientMsgProtobuf<SteammessagesClientserver2.CMsgClientPurchaseResponse.Builder> purchaseResponse;
        purchaseResponse = new ClientMsgProtobuf<>(SteammessagesClientserver2.CMsgClientPurchaseResponse.class, packetMsg);
        getClient().postCallback(new PurchaseResponseCallback(purchaseResponse.getTargetJobID(), purchaseResponse.getBody()));
    }
}

File: app/src/main/java/com/steevsapps/idledaddy/handlers/PurchaseResponse.kt
package com.steevsapps.idledaddy.handlers

import com.steevsapps.idledaddy.handlers.callbacks.PurchaseResponseCallback
import `in`.dragonbra.javasteam.base.ClientMsgProtobuf
import `in`.dragonbra.javasteam.base.IPacketMsg
import `in`.dragonbra.javasteam.enums.EMsg
import `in`.dragonbra.javasteam.handlers.ClientMsgHandler
import `in`.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserver2.CMsgClientPurchaseResponse

class PurchaseResponse : ClientMsgHandler() {

    override fun handleMsg(packetMsg: IPacketMsg) {
        when (packetMsg.msgType) {
            EMsg.ClientPurchaseResponse -> handlePurchaseResponse(packetMsg)
            else -> Unit
        }
    }

    private fun handlePurchaseResponse(packetMsg: IPacketMsg) {
        val purchaseResponse = ClientMsgProtobuf<CMsgClientPurchaseResponse.Builder>(
            CMsgClientPurchaseResponse::class.java,
            packetMsg
        )

        PurchaseResponseCallback(
            purchaseResponse.targetJobID,
            purchaseResponse.body
        ).also(getClient()::postCallback)
    }
}
